packages/ifpack2/test/unit_tests/Ifpack2_UnitTestOverlappingRowMatrix.cpp

```diff
@@ -416,9 +556,82 @@ TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL(Ifpack2OverlappingRowMatrix, getLocalDiag, Sca
     TEST_EQUALITY(ldGids[i],ovrmGids[i]);
 }
 
+
+TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL(Ifpack2OverlappingRowMatrix, reducedMatvec, Scalar, LocalOrdinal, GlobalOrdinal)
+{
+  using SC = Scalar;
+  using LO = LocalOrdinal;
+  using GO = GlobalOrdinal;
+  using NO = Tpetra::Map<>::node_type;
+  using map_type = Tpetra::Map<LO,GO,NO>;
+  using row_matrix_type =  Tpetra::RowMatrix<SC,LO,GO,NO>;
+  using MV = Tpetra::MultiVector<SC,LO,GO,NO>;
+  using Teuchos::RCP;
+
+  Tpetra::global_size_t num_rows_per_proc = 5;
+  
+  const RCP<const Tpetra::Map<LocalOrdinal,GlobalOrdinal,Node> > rowmap = tif_utest::create_tpetra_map<LocalOrdinal,GlobalOrdinal,Node>(num_rows_per_proc);
+  
+  RCP<const Tpetra::CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> > A = tif_utest::create_test_matrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>(rowmap);
+
+  // This needs to be one less than the number of matvecs we test
+  int overlapLevel = 2;
+  Ifpack2::OverlappingRowMatrix<row_matrix_type> ovA(A, overlapLevel);
+
+  RCP<const row_matrix_type> ExtMatrix = ovA.getExtMatrix();
+  SC one = Teuchos::ScalarTraits<SC>::one(), zero = Teuchos::ScalarTraits<SC>::zero();
+
+  // Vectors in the non-overlapping space
+  int numVecs = 2; 
+  MV x(rowmap,numVecs), y_direct(rowmap,numVecs), y_overlap(rowmap,numVecs);
+  x.putScalar(one); 
+
+  {
+    // Direct approach
+    MV temp1(rowmap,numVecs), temp2(rowmap,numVecs);
+    A->apply(x,temp1);
+    A->apply(temp1,temp2);
+    A->apply(temp2,y_direct);
+  }
+
+  {
+    // Overlap approach
+    RCP<const map_type> ovRowmap = ovA.getRowMap();
+    MV ovX(ovRowmap,numVecs), ovY(ovRowmap,numVecs), temp1(ovRowmap,numVecs), temp2(ovRowmap,numVecs);
+    ovX.putScalar(zero);
+    Teuchos::ArrayView<size_t> hstarts = ovA.getExtHaloStarts();
+    ovA.importMultiVector(x,ovX);
+
+    printf("Halo Starts:");
+    for(size_t i=0; i< (size_t)hstarts.size(); i++)
+      printf("%d ",(int) hstarts[i]);
+    printf("\n");
+    
+    reducedMatvec(ovA,ovX,hstarts[2],temp1);
+    reducedMatvec(ovA,temp1,hstarts[1],temp2);
+    reducedMatvec(ovA,temp2,hstarts[0],ovY);
+
+    // This isn't a Kokkos::deep_copy() since the Kokkos view was complaining about invalid ranges
```

Comment:
Could you please clarify?

Reply:
Kokkos::subview woofs in a static assert.  Something about it not liking the ranges.  Wasn't obvious why.

Reply:
Could you post the error message here?  Let's see if we can figure it out....

Reply:
Code:
```
 auto ovY_lcl = ovY.getLocalViewDevice();
    auto Y_lcl = y_overlap.getLocalViewDevice();
    auto ovYsub = Kokkos::subview(ovY_lcl,std::make_pair(0,Y_lcl.extent(0)), Kokkos::ALL);
```

Error Message:
```
/ascldap/users/csiefer/Trilinos/sandbox/Trilinos/packages/kokkos/core/src/impl/Kokkos_ViewMapping.hpp: In instantiation of ‘struct Kokkos::Impl::is_integral_extent<0u, std::pair<int, long unsigned int>, Kokkos::Impl::ALL_t>’:
/ascldap/users/csiefer/Trilinos/sandbox/Trilinos/packages/kokkos/core/src/impl/Kokkos_ViewMapping.hpp:3230:5:   required from ‘class Kokkos::Impl::ViewMapping<void, Kokkos::ViewTraits<double**, Kokkos::LayoutLeft, Kokkos::Serial, void>, std::pair<int, long unsigned int>, Kokkos::Impl::ALL_t>’
/ascldap/users/csiefer/Trilinos/sandbox/Trilinos/packages/kokkos/core/src/Kokkos_View.hpp:2487:1:   required by substitution of ‘template<class D, class ... P, class ... Args> typename Kokkos::Impl::ViewMapping<void, Kokkos::ViewTraits<DataType, Properties ...>, Args ...>::type Kokkos::subview(const Kokkos::View<D, P ...>&, Args ...) [with D = double**; P = {Kokkos::LayoutLeft, Kokkos::Serial, void}; Args = {std::pair<int, long unsigned int>, Kokkos::Impl::ALL_t}]’
/ascldap/users/csiefer/Trilinos/sandbox/Trilinos/packages/ifpack2/test/unit_tests/Ifpack2_UnitTestOverlappingRowMatrix.cpp:617:89:   required from ‘void {anonymous}::Ifpack2OverlappingRowMatrix_reducedMatvec_UnitTest<Scalar, LocalOrdinal, GlobalOrdinal>::runUnitTestImpl(Teuchos::FancyOStream&, bool&) const [with Scalar = double; LocalOrdinal = int; GlobalOrdinal = int; Teuchos::FancyOStream = Teuchos::basic_FancyOStream<char>]’
/ascldap/users/csiefer/Trilinos/sandbox/Trilinos/packages/ifpack2/test/unit_tests/Ifpack2_UnitTestOverlappingRowMatrix.cpp:646:1:   required from here
/ascldap/users/csiefer/Trilinos/sandbox/Trilinos/packages/kokkos/core/src/impl/Kokkos_ViewMapping.hpp:315:3: error: static assertion failed: subview argument must be either integral or integral extent
   static_assert( value ||

```

Reply:
Figured it out.  Kokkos::subview is deathly alergic to size_t's in their pair type...

Reply:
std::make_pair(0, (any size_t variable)) ends in suffering.

Reply:
@csiefer2 Right -- `std::make_pair(0, size_t(k))` would deduce a return type of `std::pair<int, size_t>`.  `Kokkos::subview` doesn't seem to know what to do with that....